Compensation integration enhancements

lymanepp opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
I don't believe this is a breaking change as existing entities should continue to use their original unique_id (if any) and name.

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Configuration enhancements.
1. ~~Default unique_id if not configured~~
2. Added optional configuration of sensor name.
3. ~~Generate cleaner default name when optional sensor name not configured (# 1).~~
4. Added optional configuration of sensor `device_class` ~~and `icon`~~.
5. Copy `device_class` from source when not using `attribute` as the source and optional `device_class` config is not present (# 3). ~~Same logic for `icon`.~~

General maintenance:
1. Replace deprecated numpy methods with their replacement.
2. Use _attr_xxx attributes to replace unique_id, name, should_poll, native_value, extra_state_attributes and native_unit_of_measurement methods.
3. Add type hints.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: N/A
- This PR is related to issue: N/A
- Link to documentation pull request: N/A

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [X] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

lymanepp committed:
* Default unique_id if not configured
* Optional configuration of sensor name
* Cleaner default name when optional config not present
* Optional configuration of sensor device_class
* Copy device_class from source when not using 'attribute' source and optional device_class config is not present
* Copy unit_of_measurement from source when not using 'attribute' source and optional unit_of_measurement config is not present
* Replace deprecated numpy methods with their replacement
* Use _attr_xxx attributes to replace unique_id, name, should_poll, native_value, extra_state_attributes and native_unit_of_measurement methods

lymanepp committed:
Fixed tests

lymanepp committed:
Update tests

lymanepp commented on tests/components/compensation/test_sensor.py:
This error doesn't happen when using `Polynomial.fit`

Petro31 commented on homeassistant/components/compensation/sensor.py:
move this inside try as it's out used outside the try.  Also, assign it None.

Petro31 commented on tests/components/compensation/test_sensor.py:
What happens with this data set?  What poly is returned?

lymanepp commented on tests/components/compensation/test_sensor.py:
The coefficients are `[1.0, 0.0]` for this test and the polynomial always returns `1`.

lymanepp committed:
Code review feedback

lymanepp commented on homeassistant/components/compensation/__init__.py:
Fixed

lymanepp committed:
Fix tests

lymanepp commented on homeassistant/components/compensation/sensor.py:
Cleaned this up

lymanepp commented on tests/components/compensation/test_sensor.py:
Added assertion when "unique_id" isn't configured. The entity will not be registered and configurable in the UI.

lymanepp committed:
Cleanup

lymanepp commented:
@Petro31, since I'm a first-time contributor, would you kindly approve running workflows?

lymanepp committed:
Added icon configuration

elupus commented:
Name and icon can be set from UI , once they have unique id. So should not be part of yaml config.

lymanepp committed:
Remove name and icon configuration

lymanepp commented:
> Name and icon can be set from UI , once they have unique id. So should not be part of yaml config.

Okay. Together with removing the default `unique_id`, this change eliminates much of the benefit of this PR. The only real benefits left are adding the `domain_class` configuration (and automatic copying) along with some cleanup.

I'll probably keep using my fork of this as a custom component. Let me know if you'd like me to discard this PR.

elupus commented:
Once it's a config flow helper. Unique id is for free. 

Apart from that, why would you need the fork? Make sure it has a unique id and set those things from GUI once?

lymanepp commented:
> Apart from that, why would you need the fork? Make sure it has a unique id and set those things from GUI once?

Because requiring UI configuration for something that was in YAML configuration creates a bad user experience. See my real-life config above.

elupus commented:
I can see your point in that. But your yaml doesn't set any name or device class or icon either. So i dont get your point. Sane defaults always make sense, such as inheriting the source data.

This integration will be moving to being a fully UI component at some not too far future point.

elupus commented:
Right. So why not set sane names in this pull request? The only thing to keep in mind is to keep backward compat. So old entities retain their old name/entity. Easiest being making sure that the unique_id it previously had is retained. (did it have unique id?)

lymanepp commented:
> Right. So why not set sane names in this pull request? The only thing to keep in mind is to keep backward compat. So old entities retain their old name/entity. Easiest being making sure that the unique_id it previously had is retained. (did it have unique id?)

I started with that and it was removed due to backward compatibility which will prevent any meaningful improvements in the configuration.

Petro31 commented on tests/components/compensation/test_sensor.py:
Ok, that works

Petro31 commented:
@elupus I don't see this integration ever making it to the UI.  Until config flow allows for unlimited listed fields, this will never be in the UI.  This was something I couldn't get around when I originally wrote this.

With that being said, I think we should allow name to be configured via YAML.  There are plenty of other YAML only configurations that allow this, I don't see why we can't allow it here.  As for icon, that should be set by device_class and it shouldn't be a configuration option.

lymanepp committed:
Restore name configuration

lymanepp commented:
> With that being said, I think we should allow name to be configured via YAML. There are plenty of other YAML only configurations that allow this, I don't see why we can't allow it here.

I restored the name configuration option. Would you kindly enable workflows for this PR?

Petro31 commented:
> > With that being said, I think we should allow name to be configured via YAML. There are plenty of other YAML only configurations that allow this, I don't see why we can't allow it here.
> 
> I restored the name configuration option. Would you kindly enable workflows for this PR?

Sorry, I don't have the ability to do that.  We have to wait for a core member.

lymanepp commented:
It looks like workflows are enabled now. Are there any other changes you'd like to see made on this PR?

lymanepp committed:
Don't catch exceptions that can't be raised

lymanepp commented on homeassistant/components/compensation/__init__.py:
This exception is not possible

lymanepp committed:
Add test for device_class config

lymanepp committed:
Test cleanup

lymanepp committed:
Test cleanup

lymanepp commented on homeassistant/components/compensation/sensor.py:
Ignore empty strings in addition to None

Petro31 commented:
> It looks like workflows are enabled now. Are there any other changes you'd like to see made on this PR?

I'll have to look at this when I'm back from vacation.  No access to pc atm.

Petro31 reviewed (approved):
Looks good to me.

epenet committed:
Cleanup ServiceInfo compatibility (#60540)

Co-authored-by: epenet <epenet@users.noreply.github.com>
Co-authored-by: J. Nick Koston <nick@koston.org>

Noltari committed:
Add QNAP QSW Binary Sensor platform (#70868)

frenck committed:
Merge branch 'master' into dev

balloob committed:
Allow mobile app to disable entities by default (#71562)

Noltari committed:
Add Binary Sensors for Airzone Systems (#69736)

emontnemery committed:
Add missing cast test fixture (#71595)

epenet committed:
Migrate wemo light to ColorMode (#70857)

Noltari committed:
Add Airzone to strict typing (#71604)

Noltari committed:
Add QNAP QSW to strict typing (#71603)

bdraco committed:
Fix merge conflict with master to dev in sabnzbd (CI fix) (#71605)

teharris1 committed:
Fix Insteon issue with dimmer default on level (#71426)

TheHolyRoger committed:
Add optional base64 decoding of mqtt camera image (#71223)

Add unittest for b64 decoding of camera, fix linting

mib1185 committed:
Add release url to Fritz!Tools update entity (#71606)

* add release url to update entity

* fix test

Danielhiversen committed:
Update adax lib (#71609)

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

emontnemery committed:
Add 'toggle' device action translation to fan (#71590)

emontnemery committed:
Rename logbook humanify function (#71597)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Reduce mobile app error to debug (#71601)

emontnemery committed:
Remove unneeded bind_hass from device_automation code (#71599)

frenck committed:
Update pre-commit to 2.19.0 (#71592)

Co-authored-by: Shay Levy <levyshay1@gmail.com>

bdraco committed:
Always show the start and stop event in logbook (#71600)

bdraco committed:
Make database access in the eventloop raise an exception (#71547)

bdraco committed:
Fix missing context_id in script logbook entries (#71602)

balloob committed:
Add service entity context (#71558)

Co-authored-by: Shay Levy <levyshay1@gmail.com>

andre-richter committed:
Add device_info and entity_category to Vallox (#67353)

* Add device_info and entity_category to Vallox

* Fix DeviceInfo

* Address review comments 1

* vallox suggested changes

Co-authored-by: Sebastian Lövdahl <slovdahl@hibox.fi>
Co-authored-by: J. Nick Koston <nick@koston.org>

Kane610 committed:
Bump pydeconz to v92 (#71613)

bieniu committed:
Bump nam backend library to version 1.2.4 (#71584)

Commit:
[ci skip] Translation update

bdraco committed:
Complete baked query conversion for recorder.history (#71618)

Noltari committed:
Address late QNAP QSW strict typing comments (#71628)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
Address late Airzone strict typing comments (#71627)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

emontnemery committed:
Check state attributes in template light tests (#71608)

Shutgun committed:
Bump devolo-plc-api to 0.8.0 (#71633)

emontnemery committed:
Fix issue creation links in log messages (#71638)

bdraco committed:
Convert logbook to use lambda_stmt (#71624)

frenck committed:
Fix Plugwise recovering from aiohttp client error (#71642)

lymanepp committed:
Merge branch 'dev' into compensation-enhancements

lymanepp committed:
Set device_class and unit_of_measurement at startup

lymanepp committed:
Set device_class and unit_of_measurement at startup

Noltari committed:
Update aioqsw to 0.0.8 (#71640)

Implements PEP 561, which allows to remove coordinator cast.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Noltari committed:
Update aioairzone to 0.4.4 (#71641)

Implements PEP 561, which allows to remove coordinator cast for strict typing.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

rappenze committed:
Fix wrong brightness level change visible in UI (#71655)

grablair committed:
Change Ring Chime play sound Buttons to a Siren (#71449)

Adminiuga committed:
Bump up ZHA dependencies (#71663)

Commit:
[ci skip] Translation update

dacwe committed:
Bump yalexs to 1.1.25 for handling locks in "secure locked mode" as locked (#71666)

emontnemery committed:
Optimistically set hs_color in template light (#71629)

* Optimistically set hs_color in template light

* Update light.py

* Update test

bouwew committed:
Refactor Plugwise select and add regulation_mode selector (#69210)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bdraco committed:
Add additional test cover for history_stats (#71648)

Shutgun committed:
Bump devolo_home_control (#71639)

raman325 committed:
Improve eight sleep code quality and fix bug (#71622)

bdraco committed:
Bump flux_led to 0.28.29 (#71665)

Kane610 committed:
Streamline setup of deCONZ fan platform (#71658)

Kane610 committed:
Streamline setup of deCONZ cover platform (#71656)

Kane610 committed:
Streamline setup of deCONZ lock from light platform (#71659)

Kane610 committed:
Streamline setup of deCONZ siren platform (#71660)

Kane610 committed:
Streamline setup of deCONZ switch platform (#71661)

gjohansson-ST committed:
Sensibo test data adjustment from late review (#71482)

* Sensibo test data

* Separate pytest fixture for json load

* Scope to session

trvrnrth committed:
Allow RTMP sources for RTSPtoWeb (#71695)

bdraco committed:
Significantly improve logbook performance when selecting entities (#71657)

bdraco committed:
Add history/history_during_period websocket endpoint (#71688)

nicjansma committed:
Add missing Coinbase RATEs (#65101)

frenck committed:
Remove YAML configuration from International Space Station (ISS) (#71693)

bdraco committed:
Fix sqlalchemy warning about logbook query being converted from subquery (#71710)

gwww committed:
ElkM1 integration updates for new version of base library (#71508)

Co-authored-by: J. Nick Koston <nick@koston.org>

raman325 committed:
Fix zwave_js device automation bug (#71715)

bdraco committed:
Add logbook/get_events websocket endpoint (#71706)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Prevent history_stats from rejecting states when microseconds differ (#71704)

bdraco committed:
Add device_id and logbook descriptions to lutron_caseta (#71713)

bdraco committed:
Update sql to prepare for sqlalchemy 2.0 (#71532)

* Update sql to prepare for sqlalchemy 2.0

* config flow as well

raman325 committed:
Set PARALLEL_UPDATES to 0 for all zwave_js platforms (#71626)

raman325 committed:
Enable sentry reporting for zwave_js (#71719)

* Enable sentry reporting for zwave_js

* Fix test

raman325 committed:
Send initial message for certain zwave_js ws subscriptions (#71723)

* Send initial message for certain zwave_js ws subscriptions

* Be consistent

* fix tests and bugs

raman325 committed:
Parallelize zwave_js service calls (#71662)

* Parallelize zwave_js service calls to speed them up and handle exceptions properly

* Fix bug

* Add tests

* Fix comments

* Additional comment fixes

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Remove YAML configuration from DuneHD (#71694)

frenck committed:
Remove YAML configuration from filesize (#71692)

Sholofly committed:
Add Geocaching integration (#50284)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Reinder Reinders <reinder.reinders@gmail.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>

frenck committed:
Remove username entity from Geocaching (#71728)

jesserockz committed:
Add amperage limit number to JuiceNet (#71716)

* Add amperage limit number to JuiceNet

* coverage exception

* Use mixin dataclass

emontnemery committed:
Tweak template alarm tests (#71730)

epenet committed:
Use HVACAction in mqtt (#71726)

bouwew committed:
Bump plugwise to v0.18.2 (#71731)

emontnemery committed:
Tweak template cover tests (#71732)

emontnemery committed:
Remove unused calls fixture from template tests (#71735)

marvinroger committed:
Fix timezone issue on onvif integration (#70473)

emontnemery committed:
Tweak template lock tests (#71734)

emontnemery committed:
Tweak template switch tests (#71738)

emontnemery committed:
Remove prints from template tests (#71746)

natekspencer committed:
Refactor litterrobot to use SensorEntityDescription (#71224)

gjohansson-ST committed:
Improve code quality in sql integration (#71705)

uvjustin committed:
Add use_wallclock_as_timestamps option to generic (#71245)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

PaulAnnekov committed:
Changed API for Ukraine Alarm (#71754)

bdraco committed:
Use ciso8601 for parsing datetimes with sqlalchemy sqlite dialect (#71766)

bdraco committed:
Fix process_datetime_to_timestamp and add test coverage (#71755)

bdraco committed:
Add additional context data to logbook events (#71721)

Commit:
[ci skip] Translation update

bdraco committed:
Restore v23 stats migration tests (#71743)

allenporter committed:
Prepare google calendar integration for Application Credentials (#71748)

* Prepare google calendar integration for Application Credentials

Update google calendar integration to have fewer dependencies on
yaml configuration data to prepare for supporting application
credentials, which means setup can happen without configuration.yaml
at all. This pre-factoring will allow the following PR adding
application credentials support to be more focused.

* Add test coverage for device auth checks

J3173 committed:
Create Update entities for Ubiquiti network devices (#71700)

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

Kane610 committed:
Streamline setup of deCONZ climate platform (#71708)

frenck committed:
Update coverage to 6.3.3 (#71772)

frenck committed:
Remove deprecated WLED update button (#71775)

Kane610 committed:
Streamline setup of deCONZ lock from sensor platform (#71707)

epenet committed:
Adjust pylint plugin for climate HVACAction (#70760)

emontnemery committed:
Migrate limitlessled light to color_mode (#69430)

frenck committed:
Remove deprecated Raspberry Pi GPIO integration (#71777)

iMicknl committed:
Add deprecation warning to Somfy integration (#71653)

emontnemery committed:
Support this variable in template alarm actions (#71744)

emontnemery committed:
Tweak template light tests (#71729)

tetienne committed:
Don’t send None value within Command parameter value in Overkiz integration (#71582)

Djelibeybi committed:
Remove LIFX bulb discovery from the inflight list if it fails to connect (#71673)

Remove the bulb discovery from the inflight list if it fails to connect

Signed-off-by: Avi Miller <me@dje.li>

frenck committed:
Update jinja2 to 3.1.2 (#71780)

emontnemery committed:
Support this variable in template switch actions (#71799)

emontnemery committed:
Support this variable in template button actions (#71792)

starkillerOG committed:
Fix mixing string references in Motion Blinds translations (#71806)

bdraco committed:
Update stale docstring in logbook (#71814)

bdraco committed:
Avoid buffering logbook rows unless we are selecting less than a days worth (#71809)

emontnemery committed:
Support this variable in template light actions (#71805)

emontnemery committed:
Support this variable in template vacuum actions (#71800)

emontnemery committed:
Support this variable in template select actions (#71798)

emontnemery committed:
Support this variable in template number actions (#71797)

emontnemery committed:
Support this variable in template lock actions (#71796)

emontnemery committed:
Support this variable in template fan actions (#71795)

emontnemery committed:
Support this variable in template cover actions (#71793)

emontnemery committed:
Fail template functions when no default specified (#71687)

bdraco committed:
Generate json for history and logbook websocket responses in the executor (#71813)

davet2001 committed:
Add diagnostics file export to generic camera (#71492)

emontnemery committed:
Add tilt support to Tasmota covers (#71789)

* Add tilt support to Tasmota covers

* Bump hatasmota to 0.5.0

bdraco committed:
Avoid matching entity_id/domain attributes in logbook when there is no entities_filter (#71825)

bdraco committed:
Remove unused entity_id argument in logbook context augmenter (#71829)

bdraco committed:
Remove deprecated history function entry points (#71815)

bdraco committed:
Complete refactoring of logbook humanify (#71830)

mkmer committed:
Add Aladdin connect config flow (#68304)

* Adding flow and async

* Fixes to init

* Lint and type

* Fixed coveragerc file

* Added Test Coverage

* Added Update Listener and removed unused code

* Wrong integration name in init.

* Nothing

* Added yaml import flow

* Added YAML import functionality

* Added back aladdin_connect files to coverage rc

* Removed commented code

* Clean up error message

* Update homeassistant/components/aladdin_connect/__init__.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/aladdin_connect/__init__.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Update homeassistant/components/aladdin_connect/config_flow.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

* Updated Documentation errors

* recommended change broke cover.py - backed out

* Cleaned up unused defenitions

* implimented recommended changes from gjohansson

* Dev environment cleanup

* Raised errors for better recovery, replaced removed update files, utilized PLATFORM vars to init platform

* Added back removal

* Added Code Owner

* Fixed more comment errors and import duplicates

* Added test coverage and formated code

* Added test coverage for model and init

* Added test_cover for full testing coverage

* Added await to async call

* Added missing asserts to failure tests

* Updated tranlsation

* Fixed wording in yaml import function, white space in const.py, return from validate_input.

* Update homeassistant/components/aladdin_connect/config_flow.py

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

* "too much" whitespace

* Added back mising strings.json errors

* Added ConfigFlowReconfig and tests

* Finished up reauth config flow and associated tests

* Added reauth to strings, removed username from reauth

* recommended changes, ran script.translations, added auth test to reauth

* put back self.entry.data unpack.

* Cleanup for error message, fixed missing "asserts" in tests

* Added yaml import assertions

* Fixed documentation errors in test_cover.

* remove unused string.

* revised tests and wording for yaml import

* Documentation cleanup.

* Changed sideeffect names

Co-authored-by: G Johansson <goran.johansson@shiftit.se>
Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

AngellusMortis committed:
Fix handling package detection for latest UniFi Protect beta (#71821)

Co-authored-by: J. Nick Koston <nick@koston.org>

natekspencer committed:
Add last seen and status code diagnostic sensors to litterrobot (#71760)

jjlawren committed:
Fix Sonos idle states (#71756)

epenet committed:
Adjust Fan Modes in insteon (#71804)

tkdrob committed:
Add config flow to Slack integration (#69880)

Noltari committed:
Add Sensors for Airzone WebServer (#69748)

Noltari committed:
Add QNAP QSW Button platform (#70980)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Use ciso8601 for parsing MySQLdb datetimes (#71818)

* Use ciso8601 for parsing MySQLDB datetimes

The default parser is this:

https://github.com/PyMySQL/mysqlclient/blob/5340191feb16b1f99a7b43fe7c74a2f690138cb1/MySQLdb/times.py#L66

* tweak

* tweak

* add coverage for building the MySQLdb connect conv param

gjohansson-ST committed:
Add remaining tests for Sensibo (#71764)

* Sensibo remaining tests

* Use fixture for enable entity

gjohansson-ST committed:
Add multiple departures to Trafikverket Ferry (#71484)

cliffordwhansen committed:
Update sonarr sensor (#71576)

Fixed issue when episodeFileCount/episodeCount might not exist

frenck committed:
Remove YAML configuration from Discord (#71696)

gjohansson-ST committed:
Code quality Filesize (#71768)

Commit:
Improve reliability of VLC metadata parsing (#71856)

* Improve reliability of metadata parsing.

* Remove media_album_name property

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

AngellusMortis committed:
Remove ssh switch from unsupported devices for UniFi Protect (#71859)

AlainH committed:
Update pyRFXtrx dependency to 0.29.0 (#71852)

* rfxtrx: update pyRFXtrx dependency to 0.29.0

* Update requirements_all.txt / requirements_test_all.txt

allenporter committed:
Add application credentials platform for google calendar integration (#71808)

* Add google application_credentials platform

* Further simplify custom auth implementation overrides

* Add test coverage in application_credentials

* Simplify wording in a comment

* Remove unused imports accidentally left from merge

* Wrap lines that are too long for style guide

* Move application credential loading to only where it is needed

* Leave CLIENT_ID and CLIENT_SECRET as required.

jjlawren committed:
Rework Sonos battery and ping activity tracking (#70942)

bdraco committed:
Avoid storing last_changed in the database if its the same as last_updated (#71843)

bdraco committed:
Sync event timed_fired and the context ulid time (#71854)

bdraco committed:
Small cleanups to find_next_time_expression and addition of tests (#71845)

allenporter committed:
Upgrade grpcio to 1.46.1 (#71865)

bdraco committed:
Implement use_include_order in the history websocket api (#71839)

bdraco committed:
Add MySQL index hints to logbook (#71864)

* Add MySQL index hints to logbook

* fix mysql query planner

bdraco committed:
Add Big Ass Fans integration (#71498)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

GitHub-Action committed:
[ci skip] Translation update

austinmroczek committed:
Add diagnostics for TotalConnect (#71506)

bdraco committed:
Correct typo in internal logbook function names (#71882)

starkillerOG committed:
Motion blinds restore angle (#71790)

* restore tilt for TiltDevices

* add tilt option to set_absolute_position

* improve set_absolute_position service

* fix styling

bdraco committed:
Exclude last_changed when same as last_updated for history websocket api (#71886)

bdraco committed:
Relocate sqlalchemy filter builder to recorder/filters.py (#71883)

bdraco committed:
Add logbook descriptions for mobile app zone enter and exit (#71749)

uvjustin committed:
Decouple stream options from PyAV options (#71247)

Co-authored-by: Allen Porter <allen.porter@gmail.com>

RadekHvizdos committed:
Add missing Shelly Cover sensors bugfix (#71831)

Switching Shelly Plus 2PM from switch to cover mode results in missing sensors for Power, Voltage, Energy and Temperature.

These parameters are still available in the API, but need to be accessed via "cover" key instead of "switch" key.

This change adds the missing sensors.

rappenze committed:
Limit parallel requests in fibaro light (#71762)

Kane610 committed:
Streamline setup of deCONZ number platform (#71840)

bdraco committed:
Convert history queries to use lambda_stmt (#71870)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

uvjustin committed:
Declare exports from stream explicitly (#71898)

allenporter committed:
Bump gcal_sync to 0.8.0 (#71900)

moritzbeck13 committed:
Add timer to the the helper category (#71837)

matejdro committed:
Add back description and location to calendar endpoint (#71887)

bieniu committed:
Revert changing `pysnmp` to `pysnmplib` (#71901)

moritzbeck13 committed:
Add counter to the the helper category (#71838)

GitHub-Action committed:
[ci skip] Translation update

bdraco committed:
Reduce logbook websocket payload size and parse json attributes via the DBM (#71895)

bdraco committed:
Fixing purging legacy rows and improve performance (#71916)

allenporter committed:
Add Spotify application_credentials platform (#71871)

allenporter committed:
Mark xbox configuration.yaml as deprecated after app creds import (#71908)

allenporter committed:
Improve error handling for application credentials deletion (#71868)

allenporter committed:
Add application_credentials platform to geocaching integration (#71880)

frenck committed:
Add missing title translation for the Siren domain (#71924)

jetpacktuxedo committed:
Fix VeSync air_quality fan attribute (#71771)

* Refactor attribute inclusion for VeSync fans.

A recent change to pyvesync (introduced in 2.2) changed `air_quality` to
refer to air quality as an integer representation of perceived air
quality rather than a direct reading of the PM2.5 sensor. With 2.3 the
PM2.5 sensor access was restored as `air_quality_value`. Unfortunately,
`air_quality_value` was not added as an attribute on the fan object, and
rather only exists in the `details` dictionary on the fan object.

* Update homeassistant/components/vesync/fan.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Rename `air_quality_value` attribute to `pm25`

This should make it more clear what the attribute actually represents

* `air_quality` attribute reports `air_quality_value`

This restores previous behavior for this integration to what it was
before the `pyvesync==2.02` upgrade, using the `air_quality` attribute
to report pm2.5 concentrations (formerly `air_quality`) rather the
vague measurement now reported by `air_quality`.

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ludeeus committed:
Bump awesomeversion from 22.2.0 to 22.5.1 (#71933)

chemelli74 committed:
Properly handle Shelly gen2 device disconnect (#71937)

frenck committed:
Update apprise to 0.9.8.3 (#71934)

frenck committed:
Update watchdog to 2.1.8 (#71927)

frenck committed:
Update pyupgrade to v2.32.1 (#71939)

frenck committed:
Remove auto_start translation from HomeKit (#71938)

cdce8p committed:
Update pylint to 2.13.9 (#71941)

* Update pylint to 2.13.9

* Small change

bdraco committed:
Add sensor platform to Big Ass Fans (#71877)

frenck committed:
Update PyJWT to 2.4.0 (#71928)

allenporter committed:
Update scaffold script to use application_credentials platform (#71881)

uvjustin committed:
Clean up use_wallclock_as_timestamps in generic (#71940)

dmulcahey committed:
Add additional configuration entities for ZHA lights (#70597)

* Add more configuration entities for ZHA lights

* fix typing circular imports

* enhance filter in entity factory

* fix read attribute chunking

* add test

* add exception test

frenck committed:
Update twentemilieu to 0.6.1 (#71953)

bdraco committed:
Guard expensive `cast`s in performance sensitive spots with `if TYPE_CHECKING` (#71960)

bramkragten committed:
Update frontend to 20220516.0 (#71964)

frenck committed:
Update sentry-sdk to 1.5.12 (#71930)

* Update sentry-sdk to 1.5.12

* Remove now unneeded mypy ignore

Kane610 committed:
Streamline setup of deCONZ binary sensor platform (#71820)

bdraco committed:
Guard against recorder pool current connection disappearing during global destruction (#71971)

bdraco committed:
Remove unnecessary flush from recorder (#71910)

jjlawren committed:
Add a timeout during Sonos speaker setup (#71973)

bdraco committed:
Use async_capture_events for core tests (#71970)

ollo69 committed:
Enable NUT strict typing (#71913)

bdraco committed:
Bump unifi-discovery to 1.1.3 (#71975)

balloob committed:
Inverse parallel updates default check, follow sync "update" method (#71720)

GitHub-Action committed:
[ci skip] Translation update

AngellusMortis committed:
Refactor button code to allow for other button types for UniFi Protect (#71911)

Co-authored-by: J. Nick Koston <nick@koston.org>

uvjustin committed:
Reuse codec_context on stream thread restart (#71942)

frenck committed:
Update dsmr_parser to v0.33 (#71946)

allenporter committed:
Add application_credentials platform for netatmo (#71884)

bdraco committed:
Add support for specifying the integrations manifest/list fetches (#71982)

* Add support for specifying the integrations manifest/list fetches

See https://github.com/home-assistant/core/pull/71979 for
the motovation

* Update homeassistant/components/websocket_api/commands.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Support requesting translations for multiple integrations in a single request (#71979)

elupus committed:
Avoid polling fjaraskupan if no broadcast is received (#71969)

* Switch to subclassed coordinator

* Avoid polling fjäråskupan only do on request

We still want to timeout the availability if no data was received.

* Remove unused variable

* Update homeassistant/components/fjaraskupan/__init__.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

matrixd2 committed:
Add YoLink product integration (#69167)

* add yolink integration with door sensor

* Add test flow and update .coveragerc

* Yolink integration Bug fix

* resovle test flow

* issues resolve

* issues resolve

* issues resolve

* resolve issues

* issues resolve

* issues resolve

* change .coveragerc and test_flow

* Update yolink api version

* add test for config entry

* change config flow and add test cases

* remove config entry data

* Add token check for re-auth test flow

* fix test flow issues

* Add application credentials

* Add alias for application_credentials

* support application credentials and cloud account linking

* fix suggest change

allenporter committed:
Add Home Connect application_credentials platform and deprecate configuration.yaml (#71988)

Add Home Connect application_credentials platform

ollo69 committed:
Improve NUT typing (#72002)

frenck committed:
Clean up use of deprecated async_get_registry methods (#72001)

rhadamantys committed:
Provide unique id for enocean devices (#71774)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Fix clear config entry from registry in Samsung TV migration (#72004)

* Fix clear config entry from device registry

* Fix clear config entry from entity registry

farmio committed:
Update xknx to 0.21.3 (#72006)

frenck committed:
Add more to no implicit reexport modules (#71947)

frenck committed:
Clean up accessing entity_registry.async_get_registry helper via hass (#72005)

epenet committed:
Fix no-implicit-reexport sorting issue (#72015)

elad-bar committed:
Add Tuya Multi-functional Sensor (dgnbj) (#71778)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

balloob committed:
Mobile app to notify when sensor is disabled (#71561)

* Mobile app to notify when sensor is disabled

* Add entity status to get_config

* Allow overriding enabled/disabled

frenck committed:
Clean up accessing dispatcher helpers via hass (#72014)

Clean up accessing ditpatcher helpers via hass

frenck committed:
Clean up accessing storage.Store helper via hass (#72009)

frenck committed:
Increase timeout for running full suite tests (#72024)

frenck committed:
Clean up accessing event helpers via hass (#72011)

balloob committed:
Make sure empty get_events results is always a list (#72021)

frenck committed:
Clean up accessing service helpers via hass (#72013)

thecode committed:
Clean up Shelly async methods that are not awaiting (#72026)

frenck committed:
Clean up accessing intent helpers via hass (#72028)

lymanepp committed:
Merge branch 'compensation-enhancements' of https://github.com/lymanepp/ha-core into compensation-enhancements

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#69888).

Once you do that we will be able to review and accept this pull request.

Thanks!

bachya commented:
@lymanepp Please rebase this PR off of the `dev` branch so it only includes your changes.
